distributed/comm/tests/test_comms.py

```diff
@@ -457,6 +456,39 @@ async def test_inproc_specific_different_threads():
     await check_inproc_specific(run_coro_in_thread)
 
 
+@pytest.mark.asyncio
+async def test_inproc_continues_listening_after_handshake_error():
+    async def handle_comm():
+        pass
+
+    async with listen("inproc://", handle_comm) as listener:
+        addr = listener.listen_address
+        scheme, loc = parse_address(addr)
+        connector = get_backend(scheme).get_connector()
+
+        comm = await connector.connect(loc)
+        await comm.close()
+
+        comm = await connector.connect(loc)
+        await comm.close()
```

Comment:
Is there actually an error somewhere?

Reply:
"Connection closed before handshake completed" will be logged

Reply:
Ah right. The error is that we're not engaging in the handshake. Thanks